coretex/entities/task_run/parameter/utils.py

```diff
@@ -65,3 +67,36 @@ def validateRangeStructure(name: str, value: Dict[str, Any], required: bool) ->
         return False, "Range parameter does not support float values"
 
     return True, None
+
+def getParamTypeByValueType(value: Any, name: str) -> ParameterType:
+    if isinstance(value, int):
+        return ParameterType.integer
+
+    if isinstance(value, float):
+        return ParameterType.floatingPoint
+
+    if isinstance(value, bool):
+        return ParameterType.boolean
+
+    if isinstance(value, str):
+        return ParameterType.string
+
+    if isinstance(value, list):
+        return getParamTypeByListValueType(value, name)
+
+    supportedTypes = [type_.name for type_ in ParameterType]
+
+    raise RuntimeError(f">> [Coretex] Parameter \"{name}\" has invalid type. Expected \"{supportedTypes}\", got \"{type(value)}\".")
+
+def getParamTypeByListValueType(value: List[Any], name: str) -> ParameterType:
+    if all(isinstance(item, int) for item in value):
+        return ParameterType.intList
+
+    if all(isinstance(item, float) for item in value):
+        return ParameterType.floatList
+
+    if all(isinstance(item, str) for item in value):
+        return ParameterType.strList
+
+    types_found = ", ".join({type(item).__name__ for item in value})
```

Comment:
Use `[]` brackets instead of `{}`. `{}` is used for dict comprehension whilst `[]` is used for list comprehension.

Reply:
Use `camelCase` not `snake_case`